diff --git a/services/bsky/api.ts b/services/bsky/api.ts
index 19287a32b..e54648c6e 100644
--- a/services/bsky/api.ts
+++ b/services/bsky/api.ts
@@ -1,18 +1,66 @@
 import assert from 'node:assert'
 import cluster from 'node:cluster'
-import { BskyAppView, ServerConfig } from '@atproto/bsky'
+import * as bsky from '@atproto/bsky' // import all bsky features
 import { Secp256k1Keypair } from '@atproto/crypto'
 
 const main = async () => {
   const env = getEnv()
-  const config = ServerConfig.readEnv()
+  const config = bsky.ServerConfig.readEnv()
   assert(env.serviceSigningKey, 'must set BSKY_SERVICE_SIGNING_KEY')
   const signingKey = await Secp256k1Keypair.import(env.serviceSigningKey)
-  const bsky = BskyAppView.create({ config, signingKey })
-  await bsky.start()
+
+// starts: involve logics in packages/dev-env/src/bsky.ts >>>>>>>>>>>>>
+// Separate migration db in case migration changes some connection state that we need in the tests, e.g. "alter database ... set ..."
+  const migrationDb = new bsky.Database({
+      url: env.dbPostgresUrl,
+      schema: env.dbPostgresSchema,
+    })
+  if (env.migration) {
+      await migrationDb.migrateToOrThrow(env.migration)
+  } else {
+      await migrationDb.migrateToLatestOrThrow()
+  }
+  await migrationDb.close()
+
+  const db = new bsky.Database({
+      url: env.dbPostgresUrl,
+      schema: env.dbPostgresSchema,
+      poolSize: 10,
+  })
+
+  const dataplane = await bsky.DataPlaneServer.create(
+      db,
+      env.dataplanePort,
+      config.didPlcUrl,
+      // Resolves against an in-process PLC on localhost.
+      globalThis.fetch,
+  )
+
+  const bsyncSub = new bsky.BsyncSubscription({ config, db, })
+
+
+// ends: involve logics in packages/dev-env/src/bsky.ts   <<<<<<<<<<<<<
+
+  const server = bsky.BskyAppView.create({ config, signingKey })
+// starts: involve logics in packages/dev-env/src/bsky.ts >>>>>>>>>>>>>
+  const sub = new bsky.RepoSubscription({
+      service: env.repoProvider,
+      db,
+      idResolver: dataplane.idResolver,
+      background: new bsky.BackgroundQueue(db),
+    })
+// ends: involve logics in packages/dev-env/src/bsky.ts   <<<<<<<<<<<<<
+
+  await server.start()
+  bsyncSub.start() // backports from packages/dev-env/src/bsky.ts
+  sub.start() // involve logics in packages/dev-env/src/bsky.ts
   // Graceful shutdown (see also https://aws.amazon.com/blogs/containers/graceful-shutdowns-with-ecs/)
   const shutdown = async () => {
-    await bsky.destroy()
+    await server.destroy()
+    await bsyncSub.destroy()
+    await dataplane.destroy()
+    await sub.destroy()
+    await db.close()
   }
   process.on('SIGTERM', shutdown)
   process.on('disconnect', shutdown) // when clustering
@@ -20,6 +68,11 @@ const main = async () => {
 
 const getEnv = () => ({
   serviceSigningKey: process.env.BSKY_SERVICE_SIGNING_KEY || undefined,
+  dbPostgresUrl:     process.env.BSKY_DB_POSTGRES_URL     || undefined,
+  dbPostgresSchema:  process.env.BSKY_DB_POSTGRES_SCHEMA  || undefined,
+  dataplanePort :    maybeParseInt(process.env.BSKY_DATAPLANE_PORT)     || undefined,
+  migration:         process.env.ENABLE_MIGRATIONS === 'true'           || undefined,
+  repoProvider:      process.env.BSKY_REPO_PROVIDER                     || undefined
 })
 
 const maybeParseInt = (str: string | undefined) => {
