diff --git a/.dockerbuild b/.dockerbuild
new file mode 100644
index 0000000..e69de29
diff --git a/components/repositories/DidHistory.tsx b/components/repositories/DidHistory.tsx
index 12332ac..28b54a5 100644
--- a/components/repositories/DidHistory.tsx
+++ b/components/repositories/DidHistory.tsx
@@ -1,5 +1,6 @@
 import { Loading } from '@/common/Loader'
 import { PLC_DIRECTORY_URL } from '@/lib/constants'
+import { fetchEnv } from '@/lib/envprovider'
 import { ArrowTopRightOnSquareIcon } from '@heroicons/react/20/solid'
 import { useQuery } from '@tanstack/react-query'
 
@@ -31,8 +32,8 @@ export const useDidHistory = (did: string) =>
     cacheTime: 1000 * 60 * 5,
     queryFn: async () => {
       if (!did.startsWith('did:plc')) return null
-
-      const url = `${PLC_DIRECTORY_URL}/${did}/log/audit`
+      const envs = await fetchEnv()
+      const url = new URL(`${did}/log/audit`, envs?.OZONE_DID_PLC_URL || envs?.PLC_DIRECTORY_URL || envs?.NEXT_PUBLIC_PLC_DIRECTORY_URL || 'https://plc.directory')
       const res = await fetch(url)
       return res.json()
     },
diff --git a/components/shell/LoginModal.tsx b/components/shell/LoginModal.tsx
index d76716e..4d4227c 100644
--- a/components/shell/LoginModal.tsx
+++ b/components/shell/LoginModal.tsx
@@ -7,6 +7,7 @@ import { AuthState } from '@/lib/types'
 import Client from '@/lib/client'
 import { ConfigurationFlow } from './ConfigurationFlow'
 import { ErrorInfo } from '@/common/ErrorInfo'
+import { fetchEnv } from '@/lib/envprovider'
 
 export function LoginModal() {
   const { isValidatingAuth, isLoggedIn, authState } = useContext(AuthContext)
@@ -42,6 +43,11 @@ export function LoginModal() {
     }
   }, [isLoggedIn])
 
+  useEffect(() => {
+    const pullEnv = async() => { await fetchEnv() }
+    pullEnv()
+  },[])
+
   const onSubmit = async (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault()
     e.stopPropagation()
diff --git a/lib/client.ts b/lib/client.ts
index aa799bc..89a17e8 100644
--- a/lib/client.ts
+++ b/lib/client.ts
@@ -2,6 +2,7 @@ import { AtpAgent, AtpServiceClient, AtpSessionData } from '@atproto/api'
 import { AuthState } from './types'
 import { OzoneConfig, getConfig } from './client-config'
 import { OZONE_SERVICE_DID } from './constants'
+import { fetchEnv } from '@/lib/envprovider'
 
 export interface ClientSession extends AtpSessionData {
   service: string
@@ -16,6 +17,7 @@ class ClientManager extends EventTarget {
   hasSetup = false
   private _agent: AtpAgent | undefined
   private _session: ClientSession | undefined
+  private _envs: {[key: string]: string} | undefined
 
   get authState() {
     if (!this._agent || !this._session) {
@@ -51,6 +53,7 @@ class ClientManager extends EventTarget {
     if (this.hasSetup) return this.authState
     this._session = _loadSession()
     await this._setup()
+    this._envs = await fetchEnv()// get whole envs from server side
     this.hasSetup = true
     this._emit('change')
     return this.authState
@@ -137,7 +140,7 @@ class ClientManager extends EventTarget {
   }
 
   private async _getConfig(ozoneDid?: string) {
-    const builtIn = ozoneDid || OZONE_SERVICE_DID
+    const builtIn = ozoneDid || this._envs?.OZONE_SERVER_DID || this._envs?.OZONE_SERVICE_DID || this._envs?.NEXT_PUBLIC_OZONE_SERVICE_DID || undefined
     return await getConfig(builtIn)
   }
 
diff --git a/lib/envprovider.ts b/lib/envprovider.ts
new file mode 100644
index 0000000..895bc66
--- /dev/null
+++ b/lib/envprovider.ts
@@ -0,0 +1,70 @@
+import  * as consts from '@/lib/constants'
+
+const REST_PATH_ENV = '/.env'
+
+class EnvProvider{
+    private static instance: EnvProvider = new EnvProvider()
+    private cache: {[key: string]: string} | undefined  = undefined // env holder
+
+    private constructor() {}
+    static getInstance(): EnvProvider { return EnvProvider.instance }
+
+    getEnv( key: string|null = null) {
+          if (this.cache != undefined ){
+              const envs = this.cache
+	      return (key == null) ? envs : envs[key]   // return with d
+          }
+          throw new Error("EnvProvider::getEnv(): no cache exists yet!")
+    }
+
+    async fetchEnv(
+      key: string|null = null,
+      incConst: boolean = true,
+      incBuildEnv: boolean = false,
+      restpath: string = REST_PATH_ENV,
+      save2cache: boolean = true
+    ) {
+        try {
+             return this.getEnv(key)
+        } catch (err){ /* ignore */ }
+
+        const url  = new URL(restpath, window?.location?.origin ?? '')
+        const d    = await EnvProvider._fetch(url)
+        const envs = Object.assign({},	(incConst==true)? consts: {},  (incBuildEnv==true)? process.env: {}, d ?? {} )
+        console.log("############ EnvProvider::fetchEnv => fetch ? ", (d ==undefined) ? 'failed':'success', envs )
+        if (save2cache == true) {
+          this.cache = envs
+        }
+	return (key == null) ? envs : envs[key]   // return with d
+    }
+
+    private static async _fetch(url: URL|string) {
+        try {
+            const res = await fetch(url)
+            return  await res.json()
+        } catch(err) {
+            return undefined
+        }
+    }
+}
+
+export const fetchEnv = async function(
+  key: string|null = null,
+  incConst: boolean = true,
+  incBuildEnv: boolean = false,
+  restpath: string = REST_PATH_ENV
+) {
+  return await EnvProvider.getInstance().fetchEnv(key, incConst, incBuildEnv, restpath)
+}
+
+//export const getEnv =  function( key: string|null = null) { return EnvProvider.getInstance().getEnv(key) }
+
+export const fetchEnvSync = function(
+  key: string|null = null,
+  incConst: boolean = true,
+  incBuildEnv: boolean = false,
+  restpath: string = REST_PATH_ENV
+) {
+  const pull = async () =>   { const rtn = await fetchEnv(key, incConst, incBuildEnv, restpath); return rtn};
+  return pull() // it returns Promise instance but not real value.
+}
diff --git a/lib/identity.ts b/lib/identity.ts
index 3fa4f5f..7cdb8dd 100644
--- a/lib/identity.ts
+++ b/lib/identity.ts
@@ -1,5 +1,6 @@
 import clientManager from './client'
 import { PLC_DIRECTORY_URL } from './constants'
+import { fetchEnv } from './envprovider'
 
 export const getDidFromHandle = async (
   handle: string,
@@ -20,7 +21,8 @@ export const resolveDidDocData = async function (
   did: string,
 ): Promise<DidDocData | null> {
   if (did.startsWith('did:plc:')) {
-    const url = new URL(`/${did}/data`, PLC_DIRECTORY_URL)
+    const envs = await fetchEnv()
+    const url = new URL(`/${did}/data`, envs?.OZONE_DID_PLC_URL || envs?.PLC_DIRECTORY_URL || envs?.NEXT_PUBLIC_PLC_DIRECTORY_URL || 'https://plc.directory')
     const res = await fetch(url)
     if (res.status !== 200) return null
     const doc = await res.json()
diff --git a/service/index.js b/service/index.js
index 1b7a221..0361874 100644
--- a/service/index.js
+++ b/service/index.js
@@ -31,6 +31,10 @@ async function main() {
   }
   const ozone = await OzoneService.create(config, secrets)
   // setup handlers
+
+  ozone.app.get('/.env', (_req, res) => {
+      return res.json({...process.env})
+  })
   ozone.app.get('/.well-known/ozone-metadata.json', (_req, res) => {
     return res.json({
       did: ozone.ctx.cfg.service.did,
