diff --git a/src/view/com/util/UserAvatar.tsx b/src/view/com/util/UserAvatar.tsx
index 8862bd0e4..ba3440699 100644
--- a/src/view/com/util/UserAvatar.tsx
+++ b/src/view/com/util/UserAvatar.tsx
@@ -444,9 +444,10 @@ export {PreviewableUserAvatar}
 // manually string-replace to use the smaller ones
 // -prf
 function hackModifyThumbnailPath(uri: string, isEnabled: boolean): string {
-  return isEnabled
-    ? uri.replace('/img/avatar/plain/', '/img/avatar_thumbnail/plain/')
-    : uri
+  return uri // disable hackModifyThumbnailPath
+  //return isEnabled
+    //? uri.replace('/img/avatar/plain/', '/img/avatar_thumbnail/plain/')
+    //: uri
 }
 
 const styles = StyleSheet.create({
