diff --git a/src/view/com/util/UserAvatar.tsx b/src/view/com/util/UserAvatar.tsx
index a2eba43e1..c1e34f734 100644
--- a/src/view/com/util/UserAvatar.tsx
+++ b/src/view/com/util/UserAvatar.tsx
@@ -617,7 +617,7 @@ export {PreviewableUserAvatar}
 // manually string-replace to use the smaller ones
 // -prf
 function hackModifyThumbnailPath(uri: string, isEnabled: boolean): string {
-  return isEnabled ? convertCdnPreset(uri, 'avatar_thumbnail') : uri
+  return uri // disable hackModifyThumbnailPath
 }
 
 const styles = StyleSheet.create({
