diff --git a/Dockerfile b/Dockerfile
index ed9bfad7d..195ab67e2 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -68,9 +68,6 @@ RUN \. "$NVM_DIR/nvm.sh" && \
   echo "EXPO_PUBLIC_BUNDLE_DATE=$(date -u +"%y%m%d%H")" >> .env && \
   echo "EXPO_PUBLIC_BLUESKY_PROXY_DID=$EXPO_PUBLIC_BLUESKY_PROXY_DID" >> .env && \
   echo "EXPO_PUBLIC_SENTRY_DSN=$EXPO_PUBLIC_SENTRY_DSN" >> .env && \
-  echo "##########################" && \
-  cat .env && \
-  echo "##########################" && \
   npm install --global yarn && \
   yarn && \
   yarn intl:build 2>&1 | tee i18n.log && \
@@ -95,7 +92,7 @@ RUN cd bskyweb/ && \
     -o /bskyweb \
     ./cmd/bskyweb
 
-#FROM debian:bullseye-slim
+FROM debian:bullseye-slim
 
 ENV GODEBUG=netdns=go
 ENV TZ=Etc/UTC
@@ -107,9 +104,10 @@ RUN apt-get update && apt-get install --yes \
 
 ENTRYPOINT ["dumb-init", "--"]
 
-#WORKDIR /bskyweb
-#COPY --from=build-env /bskyweb /usr/bin/bskyweb
-RUN cp -p /bskyweb /usr/bin/bskyweb
+WORKDIR /bskyweb
+COPY --from=build-env /bskyweb /usr/bin/bskyweb
+COPY --from=build-env /usr/src/social-app/bskyweb/static  /localStaticCDN/static
+COPY --from=build-env /usr/src/social-app/boot.sh         /usr/local/bin/boot.sh
 
 CMD ["/usr/bin/bskyweb"]
 
