diff --git a/bskyweb/cmd/bskyweb/server.go b/bskyweb/cmd/bskyweb/server.go
index ec5261dee..f123b8658 100644
--- a/bskyweb/cmd/bskyweb/server.go
+++ b/bskyweb/cmd/bskyweb/server.go
@@ -205,6 +205,10 @@ func serve(cctx *cli.Context) error {
 		}
 		return http.FS(fsys)
 	}())
+	// yet another static files, for localStaticCDN. it returns local file in the folder(not from embedFS bundled with bskyweb binary)
+	localStaticCDNHandler := http.FileServer(func() http.FileSystem {
+		return http.FS(os.DirFS("/localStaticCDN")) // hardcoded to reduce security risk.
+	}())
 
 	// enable some special endpoints for the "canonical" deployment (bsky.app). not having these enabled should *not* impact regular operation
 	if canonicalInstance {
@@ -224,6 +228,7 @@ func serve(cctx *cli.Context) error {
 	}
 
 	e.GET("/iframe/*", echo.WrapHandler(staticHandler))
+	e.GET("/localStaticCDN/static/*", echo.WrapHandler(http.StripPrefix("/localStaticCDN/", localStaticCDNHandler)))
 	e.GET("/static/*", echo.WrapHandler(http.StripPrefix("/static/", staticHandler)), func(next echo.HandlerFunc) echo.HandlerFunc {
 		return func(c echo.Context) error {
 			c.Response().Before(func() {
